frontend/src/scenes/instance/AsyncMigrations/AsyncMigrationDetails.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { AsyncMigration, AsyncMigrationError, asyncMigrationsLogic } from './asyncMigrationsLogic'
+import { LemonTable, LemonTableColumns } from 'lib/components/LemonTable'
+import { useActions, useValues } from 'kea'
+import { Button } from 'antd'
+import { Spinner } from 'lib/components/Spinner/Spinner'
+import { RedoOutlined } from '@ant-design/icons'
+import { humanFriendlyDetailedTime } from 'lib/utils'
+
+export function AsyncMigrationDetails({ asyncMigration }: { asyncMigration: AsyncMigration }): JSX.Element {
+    const { asyncMigrationErrorsLoading, asyncMigrationErrors } = useValues(asyncMigrationsLogic)
+    const { loadAsyncMigrationErrors } = useActions(asyncMigrationsLogic)
+
+    const columns: LemonTableColumns<AsyncMigrationError> = [
+        {
+            title: 'id',
+            dataIndex: 'id',
+        },
+        {
+            title: 'Created At',
+            render: function Render(_, asyncMigrationError: AsyncMigrationError): JSX.Element {
+                const createdAt = asyncMigrationError.created_at
+                return <div>{humanFriendlyDetailedTime(createdAt)}</div>
+            },
+        },
+        {
+            title: 'Description',
+            dataIndex: 'description',
+        },
+    ]
+    // Problem: the state is global and pressing the button below changes errors for all open pannels probably because
+    // the value asyncMigrationErrors is global and gets changed
+    // couldn't figure out how to load this on the opening of the details
```

Comment:
how should I do this?

Reply:
Currently the reducer `asyncMigrationErrors` gets its value from the result of `loadAsyncMigrationErrors`. And `loadAsyncMigrationErrors` returns data related to a specific single migration (`migrationId`), of type `AsyncMigrationError[]`.
One way this can be fixed is making `asyncMigrationErrors` a mapping based on migration IDs (`Record<AsyncMigration['id'], AsyncMigrationError[]>`). This way in `AsyncMigrationDetails` instead of `dataSource={asyncMigrationErrors}` you'd do `dataSource={asyncMigrationErrors[asyncMigration.id]}`.

Reply:
1. You can call the loading action inside a `useEffect()` when the details component renders (which equals the row having been expanded).
2. You can look at `dashboardsModel`'s `rawDashboards` reducer for inspiration, it's the same idea of memoizing with an ID-to-data object.

Reply:
I suggested earlier that Tiina can also add `onRowExpanded` to the `expandable` object, and call that (if present) from the TableRow's expansion button's `onClick` handler.

Reply:
which of those two is better?

Reply:
went with `onRowExpanded`